src/net.cpp

```diff
@@ -1923,80 +1712,74 @@ void CConnman::SocketHandler(CMasternodeSync& mn_sync)
     {
         const NodesSnapshot snap{*this, /* filter = */ CConnman::AllNodes, /* shuffle = */ false};
 
+        const auto timeout = std::chrono::milliseconds(only_poll ? 0 : SELECT_TIMEOUT_MILLISECONDS);
+
         // Check for the readiness of the already connected sockets and the
         // listening sockets in one call ("readiness" as in poll(2) or
         // select(2)). If none are ready, wait for a short while and return
         // empty sets.
-        SocketEvents(snap.Nodes(), recv_set, send_set, error_set, only_poll);
+        events_per_sock = GenerateWaitSockets(snap.Nodes());
+        if (events_per_sock.empty() || !Sock::IWaitMany(socketEventsMode, GetModeFileDescriptor(),
+                                                        ToggleWakeupPipe, timeout, events_per_sock)) {
+            if ((socketEventsMode == SocketEventsMode::Poll && !only_poll) ||
+                 socketEventsMode == SocketEventsMode::Select)
+            {
+                interruptNet.sleep_for(timeout);
```

Comment:
I think this should actually always be `SELECT_TIMEOUT_MILLISECONDS`, should not depend on `only_poll`
```suggestion
                interruptNet.sleep_for(std::chrono::milliseconds(SELECT_TIMEOUT_MILLISECONDS));
```

Reply:
The `!only_poll` condition was originally added in to make sure the timeout would always be `SELECT_TIMEOUT_MILLISECONDS` by making sure that we aren't using `timeout` when it'll be `0` (when `only_poll` is `true`) ([source](https://github.com/dashpay/dash/blob/ac867bbe17873be3b4f6e6be49753de37fb4f32a/src/net.cpp#L1641)).

It wasn't originally meant to mirror the `!only_poll` conditional sleep in `SocketEventsPoll` since the `GenerateSelectSet` was renamed to `GenerateWaitSockets` and we don't run `Sock::IWaitMany` at all if the set's empty, removing the need for a conditional timeout.

The original code did use the wait in _every_ mode (with the above condition to avoid a sleep of `0`), mirroring upstream ([source](https://github.com/bitcoin/bitcoin/blob/6e68ccbefea6509c61fc4405a391a517c6057bb0/src/net.cpp#L1460)). With that in mind, I've changed the condition to only do the sleep if we're in an LT SEM (mirroring upstream behaviour), skipping it for ET SEMs.